### PR TITLE
Ticket/3519 ncids without title style home landing page dev

### LINKED ISSUE #3492: Story: Build NCIDS Feature Card Row with Three Cards
## As a content editor, I want be able to add a Feature Card Row with Three Cards component, so that I can embed a NCIDS Feature Card row module onto Cancer.gov content.

This story is to implement one of the prioritized variants of the NCIDS Feature Card Row. For MVP2, this will be implemented on the Home & Landing page content type.

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
Scenario: [title]
  Given [context]
    And [some more context]
  When  [event]
  Then  [outcome]
    And [another outcome]
```

### Resources:
* Feature: JIRA Ticket #
* Mockups: *Here goes a URL to or the name of the mockup(s) in inVision*
* Web Page: *Here goes a URL to an existing web page*

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### LINKED ISSUE #3519: Story: NCIDS Template for Home & Landing Pages
## As a Drupal Site Builder, I want to be able to select the NCIDS Template to my Drupal site on Home & Landing pages, so that I can provide my users a way to embed and display the new NCIDS components.

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
Scenario: [title]
  Given [context]
    And [some more context]
  When  [event]
  Then  [outcome]
    And [another outcome]
```

### Resources:
InVision:
- Widescreen (1400) mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl756hwrg0282013t7d5l1urr/play
- Desktop (1024) mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl756hwrg0283013t4nhh03fd/play
- Tablet Large (880) mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl756hwrg0284013t4aj381ri/play
- Tablet (640) mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl756hwrg0285013tb131hhax/play
- Mobile Large (480) mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl756hwrg0286013tckea7wt7/play
- Mobile (320) mockup: https://publicis.invisionapp.com/console/MVP2-R2.1-Landing-Pages-cl756hway0281013t70e36jn0/cl756hwrg0287013t51y80sss/play

### Notes

*Some complementary notes if necessary:*

* > Here goes a quote from an email
* Here goes whatever useful information can exist…

---

### Prerequisites
- *required dependency for identifying the problem, scope and then solution for this issue*
- [ ] Short Spike Description - Issue #9999

### Tasks
- [ ] Short Task Description - Issue #9999
- [ ] Short Task Description - Issue #9999

### Technical Notes
- More notes, or technical nuances/direction

### LINKED ISSUE #3611: User Story: Build NCIDS Feature Card Internal
## User Story:
**As a** content editor, 
**I want to** be able to add an NCIDS Feature Card Internal to my Home and Landing page,
**so that I** can use the component to feature internal content.

**Parent Issue:** #3492 

> **ESTIMATE** TBD

### Image Sizes
Breakpoint | 1x Magnification | 1.5x Magnification | 2x Magnification
--- | --- | --- | --- 
Widescreen | 350px x 262.5px | 525px x 393.75px | 700px x 525px
Desktop |  298px x 223.5px | 525px x 393.75px |  700px x 525px
Tablet Large | 704px x 396px | 1056px x 594px | 1408px x 792px
Tablet | 504px x 283.5px | 756px x 425.25px | 1008px x 567px
Mobile Large | N/A | N/A | N/A
Mobile | N/A | N/A | N/A
    
### Acceptance Criteria

*Drupal Display:*
``` gherkin
Scenario: User can see fields for an NCIDS Feature Card Internal
  Given user is logged in to the CMS
    And user edits a "Home and Landing" content item
  When user adds an "NCIDS 3 Feature Card Row" to the "Contents" field
    And user adds an "NCIDS Feature Card Internal" to the "Feature Cards" field
  Then user will see the fields for an "NCIDS Feature Card Internal"
    And first field will be "Featured Item"
    And "Featured Item" field will be required
    And "Featured Item" field will be an entity reference to a node
    And next field will be "Override Card Title"
    And "Override Card Title" field will be optional
    And "Override Card Title" field will be a text input
    And next field will be "Override Card Description"
    And "Override Card Description" field will be optional
    And "Override Card Description" field will be a text input
    And next field will be "Override Promotional Image"
    And "Override Promotional Image" field will be optional
    And "Override Promotional Image" field will be an image select
```

*Drupal Functionality:*
``` gherkin
Scenario: User can add an NCIDS Feature Card Internal to an NCIDS 3 Feature Card Row
  Given user is logged into the CMS
  When user creates a new "Home and Landing" content item
  Then the editing page for a "Home and Landing" content item displays
    And user fills out the following fields
      | fieldLabel                  | value                                                          | field_name                     |
      | Home and Landing Page Style | NCIDS with Title                                               | field_page_style               |
      | Pretty URL                  | home-and-landing-page                                          | field_pretty_url               |
      | Page Title                  | Automated Test NCIDS Feature Card Internal                     | title                          |
      | Browser Title               | Automated Test NCIDS Feature Card Internal - Browser Title     | field_browser_title            |
      | Card Title                  | Automated Test NCIDS Feature Card Internal - Card Title        | field_card_title               |
      | Meta Description            | Automated Test NCIDS Feature Card Internal Meta Description    | field_page_description         |
      | Feature Card Description.   | Automated Test NCIDS Feature Card Internal - Feature Card Desc | field_feature_card_description |
  When user selects "Add NCIDS 3 Feature Card Row" from "Contents" dropdown
  Then "NCIDS 3 Feature Card Row" section appears
    And user fills out the following fields
      | fieldLabel | value                    | field_name              |
      | Heading    | NCIDS 3 Feature Card Row | field_container_heading |
  When user selects "Add NCIDS Feature Card Internal" from "Feature Cards" dropdown
  Then "NCIDS Feature Card Internal" section appears
    And user clicks on "Featured Item" link in the "NCIDS Feature Card Internal" area
    And user clicks on "Select content" button item
    And user selects "Article to test Related Resources" item from the list
    And user clicks on "Select content" button to select item
    And "Article to test Related Resources" had been selected
    And user fills out the following fields
      | fieldLabel                | value                                                 | field_name                      |
      | Override Card Title       | Override Card Title NCIDS Feature Card Internal       | field_override_card_title       |
      | Override Card Description | Override Card Description NCIDS Feature Card Internal | field_override_card_description |
    When user saves the content page
    Then user can see NCIDS Feature Card Internal on the page
      And card displays an image
      And card displays the title "Override Card Title NCIDS Feature Card Internal"
      And card displays the description "Override Card Description NCIDS Feature Card Internal"
```

*Front-End Field Display Rules:*
Card Title
``` gherkin
Scenario: The title set on an NCIDS Feature Card Internal is displayed
  Given user is logged in to the CMS
    And user edits a "Home and Landing" content item
    And the "Home and Landing" content item has an "NCIDS Feature Card Internal"
  When user sets the "Override Card Title" field
  Then the front-end will display the "Override Card Title" on the card
  When the user does not set the "Override Card Title"
    And the Featured Item has a "Feature Card Title"
  Then the front-end will display the "Feature Card Title" on the card
  When the user does not set the "Override Card Title"
    And the Featured Item does not have a "Feature Card Title"
  Then the front-end will display the "Browser Title" on the card
```

Card Description
``` gherkin
Scenario: The title set on an NCIDS Feature Card Internal is displayed
  Given user is logged in to the CMS
    And user edits a "Home and Landing" content item
    And "Home and Landing" content item has an "NCIDS Feature Card Internal"
  When user sets the "Override Card Description" field
  Then front-end will display the "Override Card Description" on the card
  When user does not set the "Override Card Description"
    And Featured Item has a "Feature Card Description"
  Then front-end will display the Featured Item's "Feature Card Description" on the card
  When user does not set the "Override Card Description"
    And Featured Item does not have a "Feature Card Description"
  Then front-end will display the Featured Item's "Feature Card Description" on the card
  When user does not set the "Override Card Description"
    And Featured Item does not have a "Feature Card Description"
    And Featured Item has a "List Description"
  Then front-end will display the Featured Item's "List Description" on the card
  When user does not set the "Override Card Description"
    And Featured Item does not have a "Feature Card Description"
    And Featured Item does not have a "List Description"
  Then front-end will not display a description on the card
```

Card Image
``` gherkin
Scenario: The image set on an NCIDS Feature Card Internal is displayed
  Given user is logged in to the CMS
    And user edits a "Home and Landing" content item
    And "Home and Landing" content item has an "NCIDS Feature Card Internal"
  When user has set the "Override Promotional Image"
  Then the front end will display the "Override Promotional Image" on the card
  When user has not set the "Override Promotional Image"
    And Featured Item has a "Promotional Image"
  Then the front end will display the Featured Item's "Promotional Image" on the card
  When user has not set the "Override Promotional Image"
    And Featured Item does not have a "Promotional Image"
    And the Featured Item has a "Lead Image"
  Then the front end will display the Featured Item's "Lead Image" on the card
  When user has not set the "Override Promotional Image"
    And Featured Item does not have a "Promotional Image"
    And the Featured Item does not have a "Lead Image"
  Then the placeholder image will be displayed on the card
```

``` gherkin
Scenario: The image set on an NCIDS Feature Card Internal is displayed at different aspect ratios at different breakpoints
  Given user is logged in to the CMS
    And user is viewing the home page
    And home page has an NCIDS Feature Card Internal with an image
  When the browser width is at least 880x (Widescreen, Desktop, and Tablet Large)
  Then the displayed image will have an aspect ratio of 4:3
  When the browser width is at least 640px and less than 880px (Tablet)
  Then the displayed image will have an aspect ratio of 16:9
  When the browser width is less than 640px (Mobile Large and Mobile)
  Then no image will be displayed
```

### Resources:
* Component variant design: [NCIDS #605](https://github.com/NCIOCPL/ncids/issues/605)
* **Parent**: #3492 
* **Siblings**: #3612, #3637, #3613

---

### Prerequisites
- #3551
- #3519

### Tasks
- [x] [Create NCIDS Feature Card Internal paragraph](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Paragraph-Type-NCIDS-Feature-Card#ncids-feature-card-internal)
- [x] #3690
- [x] #3691
- [x] Create basic twig templates
- [x] Evaluate component using acceptance criteria in the Parent Issue, #3492

### Technical Notes
**Content Modeling:**
- [NCIDS Feature Card Internal](https://github.com/NCIOCPL/cgov-digital-platform/wiki/paragraph-type-ncids-feature-card#ncids-feature-card-internal)
- [NCIDS 3 Feature Card Row](https://github.com/NCIOCPL/cgov-digital-platform/wiki/paragraph-type-ncids-3-feature-card-row)

**Drupal image display flow:**

image

### LINKED ISSUE #3612: User Story: Build NCIDS Feature Card External
## User Story:
**As a** content editor, 
**I want to** be able to add an NCIDS Feature Card External to my Home and Landing page,
**so that I** can use the component to feature external content.

**Parent Issue:** #3492 

> **ESTIMATE** TBD

### Image Sizes
Breakpoint | 1x Magnification | 1.5x Magnification | 2x Magnification
--- | --- | --- | --- 
Widescreen | 350px x 262.5px | 525px x 393.75px | 700px x 525px
Desktop |  298px x 223.5px | 525px x 393.75px |  700px x 525px
Tablet Large | 704px x 396px | 1056px x 594px | 1408px x 792px
Tablet | 504px x 283.5px | 756px x 425.25px | 1008px x 567px
Mobile Large | N/A | N/A | N/A
Mobile | N/A | N/A | N/A

### Acceptance Criteria
**Drupal Display:**
``` gherkin
Scenario: User can see fields for an NCIDS Feature Card External
Given user is logged in to the CMS
  And user edits a "Home and Landing" content item
When user adds an "NCIDS 3 Feature Card Row" to the "Contents" field
  And user adds an "NCIDS Feature Card External" to the "Feature Cards" field
Then user will see the fields for an "NCIDS Feature Card External"
  And first field will be "Featured Item Url"
  And "Featured Item Url" field will be required
  And "Featured Item Url" field will be a text input
  And next field will be "Card Title"
  And "Card Title" field will be required
  And "Card Title" field will be a text input
  And next field will be "Card Description"
  And "Card Description" field will be optional
  And "Card Description" field will be a text input
  And next field will be "Promotional Image"
  And "Promotional Image" field will be optional
  And "Promotional Image" field will be an image select
```

``` gherkin
Scenario: User is adding an NCIDS Feature Card External to an NCIDS 3 Feature Card Row
  Given user is logged into the CMS
  When user creates a new "Home and Landing" content item
  Then the editing page for a "Home and Landing" content item displays
    And user fills out the following fields
      | fieldLabel                  | value                                                          | field_name                     |
      | Home and Landing Page Style | NCIDS with Title                                               | field_page_style               |
      | Pretty URL                  | home-and-landing-page                                          | field_pretty_url               |
      | Page Title                  | Automated Test NCIDS Feature Card External                     | title                          |
      | Browser Title               | Automated Test NCIDS Feature Card External - Browser Title     | field_browser_title            |
      | Card Title                  | Automated Test NCIDS Feature Card External - Card Title        | field_card_title               |
      | Meta Description            | Automated Test NCIDS Feature Card External Meta Description    | field_page_description         |
      | Feature Card Description    | Automated Test NCIDS Feature Card External - Feature Card Desc | field_feature_card_description |
  When user selects "Add NCIDS 3 Feature Card Row" from "Contents" dropdown
  Then "NCIDS 3 Feature Card Row" section appears
    And user fills out the following fields
      | fieldLabel | value                    | field_name              |
      | Heading    | NCIDS 3 Feature Card Row | field_container_heading |
  When user selects "Add NCIDS Feature Card External" from "Feature Cards" dropdown
  Then "NCIDS Feature Card External" section appears
    And user fills out the following fields
      | fieldLab          | value                                   | field_name                       |
      | Featured Item Url | https://www.google.com                  | field_featured_url               |
      | Card Title        | Google Link NCIDS Feature Card External | field_override_card_title        |
      | Card Description  | Description NCIDS Feature Card External | field_override_card_description  |
    And user clicks on "Promotional Image" link in the "NCIDS Feature Card External" area
    And user clicks on "Select Image" button item
    And user selects "<Image Name>" item from the list
    And user clicks on "Select image" button to select item
    And "<Image Name>" had been selected
  When user saves the content page
    And user views the Home and Landing Page
  Then user can see NCIDS Feature Card External on the page
    And card displays an image
    And card displays the title "Google Link NCIDS Feature Card External"
    And card displays the description "Description NCIDS Feature Card External"
```

**Front-End Display Rules**

Card Description
``` gherkin
Scenario: The title set on an NCIDS Feature Card External is displayed
  Given user is logged in to the CMS
    And user edits a "Home and Landing" content item
    And "Home and Landing" content item has an "NCIDS Feature Card External"
  When user sets the "Card Description" field
  Then front-end will display the "Card Description" on the card
  When user does not set the "Card Description"
  Then front-end will not display a description on the card
```

Card Image
``` gherkin
Scenario: The image set on an NCIDS Feature Card External is displayed
  Given user is logged in to the CMS
    And user edits a "Home and Landing" content item
    And "Home and Landing" content item has an "NCIDS Feature Card External"
  When user has set the "Promotional Image"
  Then the front end will display the "Promotional Image" on the card
  When user has not set the "Promotional Image"
  Then the placeholder image will be displayed on the card
```

``` gherkin
Scenario: The image set on an NCIDS Feature Card External is displayed at different aspect ratios at different breakpoints
  Given user is logged in to the CMS
    And user is viewing the home page
    And home page has an NCIDS Feature Card External with an image
  When the browser width is at least 880x (Widescreen, Desktop, and Tablet Large)
  Then the displayed image will have an aspect ratio of 4:3
  When the browser width is at least 640px and less than 880px (Tablet)
  Then the displayed image will have an aspect ratio of 16:9
  When the browser width is less than 640px (Mobile Large and Mobile)
  Then no image will be displayed
```

### Resources:
* Component variant design: [NCIDS #605](https://github.com/NCIOCPL/ncids/issues/605)

---

### Prerequisites
- https://github.com/NCIOCPL/cgov-digital-platform/issues/3551
- https://github.com/NCIOCPL/cgov-digital-platform/issues/3519

### Tasks
- [ ] [Create NCIDS Feature Card External paragraph](https://github.com/NCIOCPL/cgov-digital-platform/wiki/paragraph-type-ncids-feature-card#ncids-feature-card-external)
- [ ] #3690
- [ ] #3691
- [ ] Evaluate component using acceptance criteria in the Parent Issue, #3492 

### Links
**Parent**: #3492 
**Siblings**: #3611, #3637, #3613

### Technical Notes
**Content Modeling:**
- [NCIDS Feature Card External](https://github.com/NCIOCPL/cgov-digital-platform/wiki/paragraph-type-ncids-feature-card#ncids-feature-card-external)
- [NCIDS 3 Feature Card Row](https://github.com/NCIOCPL/cgov-digital-platform/wiki/paragraph-type-ncids-3-feature-card-row)

**Drupal image display flow:**

image

### LINKED ISSUE #3635: Enabler: Update cgov_image with NCIDS Promo Art 4:3 override field
## Update cgov_image with NCIDS Promo Art 4:3 override field

We need to update the cgov_image media type with a new override field needed for the NCIDS Home & Landing Page components. This new NCIDS Promo Art 4:3 is needed for the feature card row in R2.1.

This issue contains the requirements for the Promo Art 4:3 and is related to the parent story #3609 .

> **ESTIMATE** TBD

### Acceptance Criteria

```gherkin
Scenario: Image editor is able to create a cgov_image and add an image to the NCIDS Promo Art 4:3 field
Given the user is logged in as an image manager
  And the user adds a new cgov_image media item
Then the cgov_image displays the NCIDS Promo Art 4:3 field
  And the field is optional
  And the field is a file upload type
  And the field shall display the tooltip text as, 

> NCIDS Promo Art 4:3 : 700x525px minimum dimensions. Keep 4:3 aspect ratio.
>       One file only.
>       10 MB limit.
>       Allowed types: png gif jpg jpeg.

When the user clicks the "Choose File" option
Then the file explorer only allows for the selection of .png, .gif, .jpg, or .jpeg files
When the user successfully uploads a file to the field
Then the Alternative Text field displays
  And the field is optional
  And the field is a text input
  And the field has the tooltip:
    Short description of the image used by screen readers and displayed when the image is not loaded. This is important for accessibility.
  And the user has the option to Remove the uploaded file.
When the user clicks Save
  And the file has dimensions less than 700px x 525px
Then the cgov_image media type is not able to be added to the system
  And an error is displayed to the user
When the user clicks Save
  And the file has dimensions of at least 700px x 525px
Then the cgov_image media type is successfully added to the system
```

**Parent:** #3609

### LINKED ISSUE #3662: Enabler: Update cgov_image with NCIDS Promo Art 16:9 Override Field
# Update cgov_image with NCIDS Promo Art 16:9 Override Field

The cgov_image media type needs to be updates with a new override field for the NCIDS Home & Landing Page components. The new media type is NCIDS Promo Art 16:9 and will support the feature card row in R2.1.

This issue contains the requirements for the Promo Art 16:9 and is related to the parent story #3609 . These requirements will be verified by the acceptance criteria contained within the parent story.

> **ESTIMATE** TBD

### Requirements

- 3609.3 The cgov_image media type shall allow the user to add an image to the optional NCIDS Promo Art 16:9 Override field.
    - 3.1 The cgov_image media type shall display the NCIDS Promo Art 16:9 Override image field.
    - 3.2 The cgov_image media type shall display the tooltip text:
      ```
      NCIDS Promo Art 16:9 : ???x???px minimum dimensions. Keep 16:9 aspect ratio.
      One file only.
      10 MB limit.
      Allowed types: png gif jpg jpeg.
      ```
    - 3.3 The cgov_image media type shall only allow file types of .png, .gif, .jpg, and .jpeg to be uploaded to the field.
    - 3.4 The cgov_image media type shall allow the user to enter Alternative Text for this field.
    - 3.5 The cgov_image media type shall allow the user to remove the image from this field.

### Parent
- #3609

### LINKED ISSUE #3776: Change Request: Using Placeholder Image on Feature Card if Images Not Uploaded
### Change Request Description
The requested change is to utilize the Placeholder Image on Feature Cards if the necessary images are not uploaded to the Card/Featured Item properly before publishing. The necessary images include the override and the cropped images for all applicable fallback images. On the Feature Card Internal and Feature Card Multimedia, the fallback images include the Override, Featured Item Promotional, and Featured Item Lead, and their respective crops. On Feature Card External, the only image is the Override Image and its crops.

### What's the expected change?
- If the override/fallback images and their crops are not properly uploaded, then the Placeholder Image will be displayed for all breakpoints.
- The guidance and tool tip will be updated to indicate to the user that if the Placeholder Image is displayed, either no image was uploaded to the Feature Card, or one of the fallback images or its crops were not properly uploaded.

### What's the current functionality?
- If the correct crop size for the breakpoint is not available, then the other sized image will appear on the Feature Card and will be the incorrect size and aspect ratio as displayed on the website.

### Updated Acceptance Criteria
Feature Card Internal
```gherkin
Scenario: The image set on an NCIDS Feature Card Internal is displayed
Given a content editor has set the required field entries in Drupal for an NCIDS Feature Card Internal
  And the content editor has saved the component in Drupal
  And the Development Solution for the component is completed in the DS
  And the Development Solution for the component has been implemented in Drupal
  And the user is viewing an NCIDS Feature Card Internal
When the content editor has set the Override Promotional Image
Then the Override Promotional Image will be displayed on the NCIDS Feature Card Internal
When the content editor has not set the Override Promotional Image
  And the Featured Item has a Promotional Image
Then the Promotional Image from the Featured Item will be displayed on the NCIDS Feature Card Internal
When the content editor has not set the Override Promotional Image
  And the Featured Item does not have a Promotional Image set
  And the Featured Item has a Lead Image
Then the Lead Image of the Featured Item will be displayed on the NCIDS Feature Card Internal
When the content editor has not set the Override Promotional Image
  And the Featured Item does not have a Promotional Image set
  And the Featured Item does not have a Lead Image set
Then the fallback image will be displayed on the NCIDS Feature Card Internal
When the content editor has not set the image uploads for the crops for their respective breakpoints
Then the fallback image will be displayed on the NCIDS Feature Card Internal at all breakpoints
```

Feature Card External
``` gherkin
Scenario: The image set on an NCIDS Feature Card External is displayed
Given a content editor has set the required field entries in Drupal for an NCIDS Feature Card External
  And the content editor has saved the component in Drupal
  And the Development Solution for the component is completed in the DS
  And the Development Solution for the component has been implemented in Drupal
  And the user is viewing the NCIDS Feature Card External
When the content editor has set the Promotional Image
Then the Promotional Image will be displayed
When the content editor has not set the Promotional Image
Then the fallback image will be displayed on the NCIDS Feature Card External
When the content editor has not set the image uploads for the crops for their respective breakpoints
Then the fallback image will be displayed on the NCIDS Feature Card External at all breakpoints
```

Feature Card Multimedia
``` gherkin
Scenario: The image set on an NCIDS Feature Card Multimedia is displayed
Given a content editor has set the required field entries in Drupal for an NCIDS Feature Card Multimedia
  And the content editor has saved the component in Drupal
  And the Development Solution for the component is completed in the DS
  And the Development Solution for the component has been implemented in Drupal
  And the user is viewing an NCIDS Feature Card Multimedia
When the content editor has set the Override Promotional Image
Then the Override Promotional Image will be displayed on the NCIDS Feature Card  Multimedia
When the content editor has not set the Override Promotional Image
  And the Featured Item has a Promotional Image
Then the Promotional Image from the Featured Item will be displayed on the NCIDS Feature Card Multimedia
When the content editor has not set the Override Promotional Image
  And the Featured Item does not have a Promotional Image set
Then the fallback image will be displayed on the NCIDS Feature Card Multimedia
When the content editor has not set the image uploads for the crops for their respective breakpoints
Then the fallback image will be displayed on the NCIDS Feature Card Multimedia at all breakpoints
```

### Additional details / screenshot
image

### Related Tickets
* Feature Card Internal #3611 
* Feature Card External #3612 
* Feature Card Multimedia #3637

### LINKED ISSUE #3779: Change Request: NCIDS Feature Cards Adjust Vertical Stack Starting Breakpoint
### Change Request Description
**Raised by Client:**
The breakpoint at which the Feature Cards in a 3 Feature Card Row switch from being side-by-side to vertically stacked is considered too high and the image is too big on the screen. The Feature Cards begin to be vertically stacked at a breakpoint less than 1024px. The request is to change the breakpoint which starts the vertical stacking from less than 1024px to less than 880px. This change request is from the client and was agreed to be implemented by the internal team on February 21, 2023.

### What's the expected change?
- The breakpoint where the Feature Cards in an NCIDS 3 Feature Card Row begin to stack vertically will start at a breakpoint less than 880px.

### What's the current functionality?
- The breakpoint where the Feature Cards in an NCIDS 3 Feature Card Row stack vertically at a breakpoint less than 1024px.

### Updated Acceptance Criteria
**Drupal Development #3611 **
``` gherkin
Scenario: The image set on an NCIDS Feature Card Internal is displayed at different aspect ratios
Given a content editor has set the required field entries in Drupal for an NCIDS Feature Card Internal
  And the content editor has saved the component in Drupal
  And the Development Solution for the component is completed in the DS
  And the Development Solution for the component has been implemented in Drupal
  And an image (Override Promotional, Featured Item Promotional, Featured Item Lead) is set for the NCIDS Feature Card Internal
  And the user is viewing the NCIDS Feature Card Internal
When the browser width is at least 880px (Widescreen, Desktop, and Tablet Large)
Then the displayed image will have an aspect ratio of 4:3
When the browser width is at least 640px and less than 880px (Tablet)
Then the displayed image will have an aspect ratio of 16:9
When the browser width is less than 640px (Mobile Large and Mobile)
Then no image will be displayed
```

**Drupal Development #3612 **
``` gherkin
Scenario: The image set on an NCIDS Feature Card External is displayed at different aspect ratios
Given a content editor has set the required field entries in Drupal for an NCIDS Feature Card External
  And the content editor has saved the component in Drupal
  And the Development Solution for the component is completed in the DS
  And the Development Solution for the component has been implemented in Drupal
  And an image (Promotional Image) is set for the NCIDS Feature Card External
  And the user is viewing the NCIDS Feature Card External
When the browser width is at least 880px (Widescreen, Desktop, and Tablet Large)
Then the displayed image will have an aspect ratio of 4:3
When the browser width is at least 640px and less than 880px (Tablet)
Then the displayed image will have an aspect ratio of 16:9
When the browser width is less than 640px (Mobile Large and Mobile)
Then no image will be displayed
```

**Drupal Development #3637 **
``` gherkin
Scenario: The image set on an NCIDS Feature Card Multimedia is displayed at different aspect ratios
Given a content editor has set the required field entries in Drupal for an NCIDS Feature Card Multimedia
  And the content editor has saved the component in Drupal
  And the Development Solution for the component is completed in the DS
  And the Development Solution for the component has been implemented in Drupal
  And an image (Override Promotional, Featured Item Promotional) is set for the NCIDS Feature Card Multimedia
  And the user is viewing the NCIDS Feature Card Multimedia
When the browser width is at least 880px (Widescreen, Desktop, and Tablet Large)
Then the displayed image will have an aspect ratio of 4:3
When the browser width is at least 640px and less than 880px (Tablet Large)
Then the displayed image will have an aspect ratio of 16:9
When the browser width is less than 640px (Mobile Large and Mobile)
Then no image will be displayed
```

### Additional details / screenshot
- Screenshot

### Related Tickets
* Original Story Ticket: NCIDS Feature Card Internal #3611 
* Original Story Ticket: NCIDS Feature Card External #3612 
* Original Story Ticket: NCIDS Feature Card Multimedia #3637

### PR DESCRIPTION
**Pull request by dev-rana-publicis**
ODE: https://ncigovcdode548.prod.acquia-sites.com/
Closes #3492 
Closes #3611 
Closes #3612 
Closes #3519 
Closes #3635 
Closes #3662 
Closes #3612 
Closes #3776 
Closes #3779

### CODE REVIEW

**Comment by andyvanavery** [comment]
During client review on Feb 21, 2023, the client indicated that the images appeared "too large" at the breakpoint where the cards vertically stack. This was submitted as a Change Request which was accepted by the internal team on Feb 21, 2023. The DS Development will execute the change in the DS. Once completed, this PR can be updated with the changes necessary to implement the different breakpoint that starts the vertical stacking.

**Comment by arcepaul** [comment]
Accessibility is where I would expect it to be (no new issues as a result of this code).  Except for some missing newlines at end of files, looks good to me so far.

**Review by sarinapadilla — APPROVED** [review]
Reviewed by clients and Product, and looks good!

**Comment by arcepaul** [comment]
Your commit message is failing commitlint because of the format of that last commit. may want to reorder and squash it into one of the other commits.

**Comment by bennettcc** [comment]
ODDC Feds have re-reviewed this and approve! (specifically, they reviewed the changes to the H2 link state changes

**Comment by andyvanavery** [comment]
The NCIDS 3 Feature Card Row ticket #3492 was updated with new Acceptance Criteria to indicate how the Heading will display above the NCIDS 3 Feature Card Row

File: docroot/profiles/custom/cgov_site/modules/custom/cgov_image/src/CgovImageGetThemeImages.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+   * Returns the extension name.
+   */
+  public function getName() {
+    return 'cgov_image.CgovImageGetThemeImages';
+  }
+
+  /**
+   * Function that returns an example text with a color passed as a parameter.
+   */
+  public function getThemeImages($image_style) {
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/modules/custom/cgov_image/src/CgovImageGetThemeImages.php:34]: change this to getPlaceholderImage.

File: docroot/profiles/custom/cgov_site/modules/custom/cgov_image/src/CgovImageGetThemeImages.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    $test = new CgovImageTools();
+    $crop = $test->findCropByStyle($image_style);
+    if ($crop === "4x3") {
+      $uri = 'module://cgov_image/img/placeholder-4x3.png';
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/modules/custom/cgov_image/src/CgovImageGetThemeImages.php:38]: Use the `$crop` variable to get the file name so this thing scales. I.E., no switch statement.

File: docroot/profiles/custom/cgov_site/modules/custom/cgov_image/src/CgovImageGetThemeImages.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    elseif ($crop === "16x9") {
+      $uri = 'module://cgov_image/img/placeholder-16x9.png';
+    }
+    $image_style = ImageStyle::load($image_style);
```
Comment (bryanpizzillo) [inline docroot/profiles/custom/cgov_site/modules/custom/cgov_image/src/CgovImageGetThemeImages.php:43]: You should see what happens when you handle a bad request.

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
I have some tweaks for the placeholders.

File: docroot/profiles/custom/cgov_site/modules/custom/cgov_image/src/CgovImageGetThemeImages.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace Drupal\cgov_image;
+
+use Drupal\image\Entity\ImageStyle;
+use Drupal\cgov_image\CgovImageTools;
+
+/**
+ * Class CgovImageGetThemeImages.
```
[the remark this answers is not in the document]
Comment (dev-rana-publicis) [reply docroot/profiles/custom/cgov_site/modules/custom/cgov_image/src/CgovImageGetThemeImages.php:9]: CgovImageTools is the twig extension I am using to get findCropByStyle. Did you still want me to change CgovImageTools to CgovImageTwigExtensions?

**Review by adrianacastaneda — APPROVED** [review]

**Pull request merged**